ch02/tacos/src/main/java/tacos/Taco.java

```diff
@@ -12,15 +12,17 @@
 public class Taco {
 
   // end::allButValidation[]
-  @NotNull
+  //@NotNull
+  @NotBlank(message="")
   @Size(min=5, message="Name must be at least 5 characters long")
   // tag::allButValidation[]
   private String name;
   // end::allButValidation[]
-  @Size(min=1, message="You must choose at least 1 ingredient")
+  @NotNull(message="You must choose at least 1 ingredient")
```

Comment:
I think it is needed to add @NotNull to ingredients, without this annotation, user can still pass the validation without choosing any ingredients, makes the @Size annotation useless